File: CSharp/modul7_kel46_queue.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Modul7_Kel46
{
    class Program
    {
        static void Main(string[] args)
        {
            int pilihan;
            int data;
            Queue antrian = new Queue();
            //looping until false condition
            do
            {
                //Displaying Menu
                Console.WriteLine("---Ini Queue-- -");
                Console.WriteLine("1. ENQUEUE Item");
                Console.WriteLine("2. DEQUEUE Item");
                Console.WriteLine("3. Lihat Daftar Data");
                Console.WriteLine("4. Lihat Data Teratas");
                Console.Write("Masukkan Pilihan : ");
                pilihan = int.Parse(Console.ReadLine());
                //condition for choice
                if (pilihan == 1)
                {
                    Console.Write("Data yang ditambahkan : ");
                    data = int.Parse(Console.ReadLine());
                    antrian.Enqueue(data);
                    Console.WriteLine("");
                }
                else if (pilihan == 2)
                {
                    if (antrian.Count != 0) {
                        antrian.Dequeue();
                        Console.WriteLine("");
                    } else {
                        Console.WriteLine("Data kosong");
                    }
                    
                }
                else if (pilihan == 3)
                {
                    foreach (int value in antrian)
                    {
                        Console.Write(value+" ");
                    }
                    Console.WriteLine("\n");
                }
                else if (pilihan == 4)
                {
                    if (antrian.Count != 0) {
                        Console.WriteLine("Data teratas : " + antrian.Peek());
                        Console.WriteLine("");
                    } else {
                        Console.WriteLine("Data kosong");
                    }
                    
                }
                else
                {
                    Console.WriteLine("Pilihan Tidak Ada!!");
                }//end of condition
            } while (pilihan != 0);//end looping
        }
     }
}

File: Python/modul7_kel46_queue.py
from collections import deque
pilihan = 0
data = 0
count = 0
queue = deque()
while count == 0:
    print("---Ini Queue---")
    print("1. Enqueue Item")
    print("2. Dequeue Item")
    print("3. Lihat Daftar Data")
    print("4. Lihat Data Teratas")
    pilihan = int(input("Masukkan Pilihan : "))
    count +=1
    if pilihan == 1:
        data = int(input("Data yang ditambahkan : "))
        queue.append(data)
        print ("")
        count = 0
    if pilihan == 2:
        if not queue :
            print("Data kosong")
            count = 0
            print ("")
        else :
            queue.popleft()
            count = 0
            print ("")
    if pilihan == 3:
        print(queue)
        count = 0
        print ("")
    if pilihan == 4:
        if not queue :
            print("Data kosong")
            count = 0
            print ("")
        else :
            print(queue[0])
            count = 0
            print ("")